Выполненый 6 спринт без тестов.

Name-Moriarty opened the pull request:
Тесты сделаю после проверки, что реализовал корректно.

Name-Moriarty committed:
Выполненый 6 спринт без тестов.

Name-Moriarty committed:
Выполненый 6 спринт без тестов.

Name-Moriarty committed:
Выполненый 6 спринт без тестов.

afomkina commented on src/history/InMemoryHistoryManager.java:
```diff
@@ -3,23 +3,62 @@
     public void add(Task task) {
-        if (taskHistory.size() == MAX_HISTORY_SIZE) {
-            taskHistory.remove(0);
+        if (taskHistory.size() == 0) {
```
Лучше использовать метод `isEmpty()`. Он проверяет, что мапа пуста и улучшает читаемость

afomkina commented on src/history/Node.java:
```diff
@@ -0,0 +1,38 @@
+        return data;
+    }
+
+    public void setData(T data) {
```
Этот сеттер нигде не используется, лучше его удалить, а поле `data` сделать `final`

afomkina reviewed (changes requested):
Не хватает части тз с тестами:

```
Поработайте над покрытием классов менеджера тестами. Отдельно протестируйте новую функциональность менеджера истории. 
Скорректируйте предыдущие тесты под изменившийся алгоритм хранения версий задачи.
Проверьте, что встроенный связный список версий, а также операции добавления и удаления работают корректно.
Покройте тестами ещё несколько функций менеджера задач, которые вы реализовали в предыдущих спринтах. Отдельно уделите внимание целостности данных:
Удаляемые подзадачи не должны хранить внутри себя старые id.
Внутри эпиков не должно оставаться неактуальных id подзадач.
С помощью сеттеров экземпляры задач позволяют изменить любое своё поле, но это может повлиять на данные внутри менеджера. Протестируйте эти кейсы и подумайте над возможными вариантами решения проблемы.
```

Name-Moriarty committed:
Написал тесты для историй и дополнительные для класса manager
Исправил  Удаляемые подзадачи не должны хранить внутри себя старые id.
Исправил  Внутри эпиков не должно оставаться неактуальных id подзадач.
Изменил метод createTask теперь он создает новый объект задачи чтобы нельзя было изменить из вне объект в менеджере

Name-Moriarty committed:
Написал тесты для историй и дополнительные для класса manager
Исправил  Удаляемые подзадачи не должны хранить внутри себя старые id.
Исправил  Внутри эпиков не должно оставаться неактуальных id подзадач.
Изменил метод createTask теперь он создает новый объект задачи чтобы нельзя было изменить из вне объект в менеджере

afomkina commented on src/manager/InMemoryTaskManager.java:
```diff
@@ -66,30 +66,33 @@ public void deleteAllSubtask() {
     @Override
-    public boolean createTask(SubTask subtask) {
+    public boolean createTask(SubTask subtaskNew) {
+        SubTask subTask = subtaskNew;
```
Нужно удалить копирование задачи во всех методах в данном классе

Name-Moriarty replied:
Удалил

afomkina commented on test/history/HistoryTest.java:
```diff
@@ -0,0 +1,67 @@
+public class HistoryTest {
+
+    private TaskManager manager = Managers.getDefault();
+    private HistoryManager historyManager = Managers.getDefaultHistory();
```
Поля 18-19 должны быть `final`

Name-Moriarty replied:
Исправил

afomkina commented on test/manager/ManagersTest.java:
```diff
@@ -6,25 +6,25 @@
 
 class ManagersTest {
     private TaskManager manager = Managers.getDefault();
     private HistoryManager historyManager = Managers.getDefaultHistory();
```
Менеджер истории не используется в данных тестах

afomkina commented on test/manager/ManagersTest.java:
```diff
@@ -6,25 +6,25 @@
 import static org.junit.jupiter.api.Assertions.*;
 
 class ManagersTest {
     private TaskManager manager = Managers.getDefault();
```
Поле должно быть `final`

afomkina commented on test/manager/ManagersTest.java:
```diff
@@ -36,14 +36,100 @@ public void createTask() {
+    }
+
+    @Test
+    public void testGetListAllTask() {
```
Что ты хотел проверить в данном тесте? Он выглядит совершенно непонятно

afomkina commented on test/manager/ManagersTest.java:
```diff
@@ -36,14 +36,100 @@ public void createTask() {
+    }
+
+    @Test
+    public void testDeleteEpic() {
```
Тест называется delete epic, но проверяется в нем удаление любых типов задач

afomkina commented on test/manager/ManagersTest.java:
```diff
@@ -36,14 +36,100 @@ public void createTask() {
+    }
+
+    @Test
+    public void testDeleteSubtask() {
```
Аналогично комментарию выше

afomkina commented on test/manager/ManagersTest.java:
```diff
@@ -36,14 +36,100 @@ public void createTask() {
+    }
+
+    @Test
+    public void testUpdateAllTask() {
```
Этот тест проверяет статусы задач, нужно поправить имя теста

afomkina commented on test/manager/ManagersTest.java:
```diff
@@ -36,14 +36,100 @@ public void createTask() {
+    }
+
+    @Test
+    public void testGetOneTask() {
```
Лучше переименовать в testGetTaskById

afomkina reviewed (changes requested):
Лучше максимально декомпозировать тесты, так как большие разнообразные проверки очень сложно поддерживать. В коммерческой разработке принято писать 1 тест на 1 тестовый случай. Например, мы хотим проверить получение эпика по id, мы напишем 1 тест, где вызовем метод  получения эпика по id и проверим, что получили именно тот эпик, который вызывали. Для получения подзадачи по id, нужно будет писать отдельный тест. Данный подход помогает быстро выявить проблему, если тесты перестали проходить и делают код более читаемым

Name-Moriarty committed:
Декомпозировал тесты, разбил их.
Исправил все замечания.
